src/components/books/AddBook.js

```diff
@@ -1,21 +1,45 @@
-// import React from "react";
+import { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { v4 as uuidv4 } from 'uuid';
+import { addBook } from '../../redux/books/books';
 
-const AddBook = () => (
-  <article>
-    <h3>ADD NEW BOOK</h3>
-    <form>
-      <input placeholder="Book title" />
-      <select placeholder="Categorie">
-        <option value="" selected disabled hidden>
-          Categorie
-        </option>
-        <option>Action</option>
-        <option>Science Fiction</option>
-        <option>Economy</option>
-      </select>
-      <button type="submit">ADD BOOK</button>
-    </form>
-  </article>
-);
+const AddBook = () => {
+  const dispatch = useDispatch();
+  const [title, setTitle] = useState('');
+
+  const handleChange = (e) => setTitle(e.target.value);
+
+  const submitBookToStore = (e) => {
+    e.preventDefault();
+    const book = {
+      id: uuidv4(),
+      title,
+    };
+
+    dispatch(addBook(book));
+    setTitle('');
+  };
+
+  return (
+    <article>
+      <h3>ADD NEW BOOK</h3>
+      <form onSubmit={submitBookToStore}>
+        <input
```

Comment:
- [ ] Every time you change the input value, that updates the local state every time. You could use `onChange` for this. Please, add this to your application. 🦊

Reply:
By using onChange instead of onSubmit is not working proper. It creates a book on every letter I am entering.

Reply:
onChange function is not in place onSubmit, is additional. The flow for this is that when you change the input value, the state inside the component change to the new value, and when you submit it, in place to take the value on input, you take the value on the state (or hooks). Make it sense to you? 😄

Reply:
I see your note, and I understand it can break the code, but is part of mandatory requirements. 
![image](https://user-images.githubusercontent.com/20563252/130665877-fe290ec6-ef6c-454e-8a3a-2c0eea696c86.png)

Reply:
onChange function is not in place onSubmit, is additional. The flow for this is that when you change the input value, the state inside the component change to the new value, and when you submit it, in place to take the value on input, you take the value on the state (or hooks). Make it sense to you? 😄

Reply:
I see your note, and I understand it can break the code, but is part of mandatory requirements. 
![image](https://user-images.githubusercontent.com/20563252/130665877-fe290ec6-ef6c-454e-8a3a-2c0eea696c86.png)

Reply:
I check the project and this change is implemented, good job!

Reply:
I check this part and the onChange part is implemented, good job! 😄